STYLEGUIDE.adoc

```diff
@@ -30,10 +25,35 @@ ____
 As opposed to "the Jenkins cluster", "agent pool" or any other phrase to
 describe the collective set of resources provided by Jenkins.
 
+
+=== Pipeline
+
+When referring to a Jenkins Pipeline via short-hand ("Pipeline"), it
+should always be title-cased. When referring to a conceptual pipeline
+("continuous delivery pipeline"), it it should always be lower-cased.
+
+With the
+link:https://jenkins.io/blog/2016/12/19/declarative-pipeline-beta/[Declarative Pipeline beta]
+there are two differen syntaxes for Pipeline which must be referred to when
+writing about the subject. When referring to the entire system, "Jenkins
+Pipeline" or "Pipeline" should be used. When referring to a specific sytnax,
+the following should be used
+
+
+Scripted Pipeline::: functionality defined by the fully-featured, expressive,
+Groovy-based DSL. Scripted Pipeline preceeded Declarative Pipeline so most
+older documentation is generally referring to Scripted Pipeline.
+
+Declarative Pipeline::: functionality provided by the Pipeline Model Definition
+plugin. Declarative Pipeline is not the fully-featured DSL but a strict subset.
```

Comment:
"Declarative Pipeline is not the fully-featured DSL but a strict subset." -> 
"Declarative Pipeline behaves much like a domain-specific language, defining a specific structure, and accept a limited subset of the Groovy language."